contracts/modules/cover/CoverNFTDescriptor.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-v4/utils/Strings.sol";
+import "@openzeppelin/contracts-v4/utils/Base64.sol";
+import "../../interfaces/ICover.sol";
+import "../../interfaces/ICoverNFTDescriptor.sol";
+import "../../libraries/DateTime.sol";
+import "./CoverNFT.sol";
+
+contract CoverNFTDescriptor {
+  using Strings for uint256;
+  using DateTime for uint256;
+
+  string[] public coverAssets;
+
+  constructor() {
+    // TODO: how to handle asset symbols?
+    coverAssets = ["WEI", "DAI"];
+  }
+
+  function tokenURI(CoverTokenURIParams calldata params) public view returns (string memory) {
+
+    string memory image = Base64.encode(bytes(generateSVGImage(params)));
+
+    return string(
+      abi.encodePacked(
+        "data:application/json;base64,",
+        Base64.encode(
+          bytes(
+            abi.encodePacked(
+              '{"name":"', params.name, '",',
+              '"description":"', generateDescription(params), '",',
+              '"image": "',
+              'data:image/svg+xml;base64,',
+              image,
+              '"}'
+            )
+          )
+        )
+      )
+    );
+  }
+
+  function generateDescription(CoverTokenURIParams calldata params) public view returns (string memory) {
+
+    CoverNFT coverNFT = CoverNFT(msg.sender);
+    ICover cover = ICover(coverNFT.operator());
+
+    if (cover.coverDataCount() < params.tokenId) {
+      return "This NFT does not exist";
+    }
+
+    // Get cover data
+    CoverData memory coverData = cover.coverData(params.tokenId);
+    string memory productName = cover.productNames(coverData.productId);
+    CoverSegment memory lastSegment = cover.coverSegments(params.tokenId, cover.coverSegmentsCount(params.tokenId) - 1);
+
+    // Get expiration timestamp and check if expired
+    uint expiryTimestamp = lastSegment.start + lastSegment.period;
+
+    // Format date to DD/MM/YYYY
```

Comment:
![image](https://user-images.githubusercontent.com/528959/219343550-e1f153f5-794a-4318-910c-cb4946e140e9.png)

Reply:
I went with `Sep 3 2023` format for readability. It's similar to the javascript datetime library `toDateString()`, just missing the day of the week. I think the day of the week is overkill though.

Reply:
yeah, sounds good 👍